src/background.js

```diff
@@ -12,27 +12,14 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-if (!document.pictureInPictureEnabled) {
-  chrome.browserAction.setTitle({ title: 'Picture-in-Picture NOT supported' });
-} else {
-  chrome.browserAction.onClicked.addListener(tab => {
-    chrome.tabs.executeScript({ file: 'script.js', allFrames: true });
+chrome.action.onClicked.addListener((tab) => {
+  chrome.storage.sync.get({ optOutAnalytics: false }, (results) => {
```

Comment:
It looks like this API returns a Promise in Manifest V3.  It would be simpler IMO to use that instead of mixing Promise-consuming and callback-passing styles.

Reply:
Thank you @mfoltzgoogle! It is much better now

Reply:
Argh. I just noticed using promises makes the extension lose the user activation. I've switched back to using callbacks to avoid this issue.